Marlin/src/HAL/shared/cpu_exception/exception_arm.cpp

```diff
@@ -0,0 +1,295 @@
+    // The frame will not be valid when returning anymore, let's clean it    
+    savedFrame.CFSR = 0;
+    
+    frame->pc = (uint32_t)resume_from_fault; // Patch where to return to
+    frame->lr = 0xdeadbeef;  // If our handler returns (it shouldn't), let's make it trigger an exception immediately
+    frame->xpsr = (1 << 24); // Need to clean the PSR register to thumb II only
+    MinSerial::force_using_default_output = true;
+    return; // The CPU will resume in our handler hopefully, and we'll try to use default serial output 
+  }
+
+  // Sorry, we need to emergency code here since the fault is too dangerous to recover from 
+  MinSerial::force_using_default_output = false;
+  resume_from_fault();
+}
+
+void hook_cpu_exceptions() {
+  // On ARM 32bits CPU, the vector table is like this:
+  // 0x0C => Hardfault
+  // 0x10 => MemFault
+  // 0x14 => BusFault
+  // 0x18 => UsageFault
+
+  // Unfortunately, it's usually run from flash, and we can't write to flash here directly to hook our instruction
+  // We could set an hardware breakpoint, and hook on the fly when it's being called, but this
+  // is hard to get right and would probably break debugger when attached
+
+  // So instead, we'll allocate a new vector table filled with the previous value except
+  // for the fault we are interested in.
+  // Now, comes the issue to figure out what is the current vector table size
+  // There is nothing telling us what is the vector table as it's per-cpu vendor specific.
+  // BUT: we are being called at the end of the setup, so we assume the setup is done
+  // Thus, we can read the current vector table until we find an address that's not in flash, and it would mark the
+  // end of the vector table (skipping the fist entry obviously)
+  // The position of the program in flash is expected to be at 0x08xxx xxxx on all known platform for ARM and the
+  // flash size is available via register 0x1FFFF7E0 on STM32 family, but it's not the case for all ARM boards
+  // (accessing this register might trigger a fault if it's not implemented).
+
+  // So we'll simply mask the top 8 bits of the first handler as an hint of being in the flash or not -that's poor and will
+  // probably break if the flash happens to be more than 128MB, but in this case, we are not magician, we need help from outside.
+
+  unsigned long * vecAddr = (unsigned long*)get_vtor();
+
+  #ifdef VECTOR_TABLE_SIZE
+    uint32_t vec_size = VECTOR_TABLE_SIZE;
+    alignas(128) static unsigned long vectable[VECTOR_TABLE_SIZE] ;
+  #else
+    #ifndef IS_IN_FLASH
+      #define IS_IN_FLASH(X) (((unsigned long)X & 0xFF000000) == 0x08000000)
+    #endif
+
+    // When searching for the end of the vector table, this acts as a limit not to overcome
+    #ifndef VECTOR_TABLE_SENTINEL
+      #define VECTOR_TABLE_SENTINEL 80
+    #endif
+
+    // Find the vector table size
+    uint32_t vec_size = 1;
+    while (IS_IN_FLASH(vecAddr[vec_size]) && vec_size < VECTOR_TABLE_SENTINEL)
```

Comment:
Why are you doing all this logic here? Is it really needed?
The method used on `DebugMonitor.cpp` (deleted here) just overwrite the handlers, like `__attribute__((naked)) void BusFault_Handler() { } `... and it works for me, at least on F4 and LPC (that I tested).

Reply:
You can't overwrite handler if they aren't weak (libmaple does not declare them weak) and if the chip flash is locked (those with a factory bootloader  usually have locked flash access) you are not allowed to write to the vector table (unless unlocking the flash, but it means erasing the complete flash, bootloader included).

Reply:
The *why* is because you can't patch flash code in software, so I need to create a new vector table in SRAM, and copy all the previous pointer to this new table and then update atomically the new vector table pointer to the SRAM version.
If we get rid of libmaple, **and** there is no other platform code that declare handler as non-weak, then we could get rid of this code completely and instead overwrite them at link time.